fix(docs-infra): fix `search-results` styles

gkalpak opened the pull request:
I started to fix the "No results found" message being hidden, but ended up doing some clean up and other minor fixes (see individual commits for details).

WRT to the main fix ("No results found" being hidden):

**BEFORE**
![search-not-found-before](https://user-images.githubusercontent.com/8604205/60532113-3ae51600-9d05-11e9-8da0-0fdc2817b653.png)

**AFTER**
![search-not-found-after](https://user-images.githubusercontent.com/8604205/60532123-40daf700-9d05-11e9-9890-80301502bbed.png)

##
Both before and after the "No results found" is displayed correctly on the "Page Not Found" page:

![page-not-found-before-and-after](https://user-images.githubusercontent.com/8604205/60532143-4c2e2280-9d05-11e9-9d9a-13791f4230d4.png)

---
**UPDATE:**

This PR also [fixes a bug][1] in `ScrollService` that caused test flakes ([example][2]). No idea why the flakes started appearing on this PR (which was just changing CSS styles 🤷‍♂).

[1]: https://github.com/angular/angular/commit/119dcd146188307f0296132536d8eb311fda96c4
[2]: https://circleci.com/gh/angular/angular/381649

gkalpak committed:
style(docs-infra): make indentation in `_search-results.scss` consistent with other `.scss` files

gkalpak committed:
refactor(docs-infra): clean up `aio-search-results` styles

gkalpak committed:
fix(docs-infra): remove redundant `search-results` styles

Previously, `.search-area` had a default style of `display: flex`, but
it was overriden to `display: block` in media queries for
`max-width: 480px` and `min-width: 600px`. As a result, it only had
`display: flex` between 481px and 599px.

Since no flex layout features are necessary inside `.search-area`, this
commit changes the style to always be `display: block` (i.e. the default
for `div` elements).

gkalpak committed:
fix(docs-infra): improve `search-results` layout on smaller screens

Previously, the `search-results` layout was switch from horizontal to
vertical at 480px. Yes, since some search queries can yield more than 5
`.search-area`s, even 600px are too narrow to accomodate a horizontal
layout.

This commit changes the breakpoint at which the layout switches to
vertical from 480px to 600px.

gkalpak committed:
fix(docs-infra): show (and style correctly) the 'No results found' message

mary-poppins commented:
You can preview 9489a5b at https://pr31390-9489a5b.ngbuilds.io/.

gkalpak committed:
test(docs-infra): clean up global listeners after `ScrollService` tests

The `ScrollService` sets up some global `window` listeners.

Previously, these listeners were never unregistered. This was not a
problem in the real app, because the `ScrollService` instance exists for
the lifetime of a user session. In tests, however, where the `window`
instance is among all tests, the listeners would survive the
`ScrollService` tests. This, in addition to the fact that we used a mock
`ViewportScroller` which did not return the expected type from
`getScrollPosition()`, caused errors to be thrown in unrelated tests
(i.e. whenever a scroll event was emitted on `window`). See [here][1]
for an example failure.

This commit fixes it by adding an `ngOnDestroy()` method that
unregisters the listeners and ensuring it is called after each
`ScrollService` test.

[1]: https://circleci.com/gh/angular/angular/381649

mary-poppins commented:
You can preview 119dcd1 at https://pr31390-119dcd1.ngbuilds.io/.

sjtrimble commented:
@gkalpak 

A couple UX/UI recommendations:
1) Don't actually show the "I have nothing to show you" message if we don't have any suggestions OR even better (maybe handle it in another PR) show them a few items where they might want to spend some time on like the blog, the guide, etc.
<img width="1409" alt="Screen Shot 2019-07-04 at 12 32 16 PM" src="https://user-images.githubusercontent.com/2958442/60685427-edc58900-9e57-11e9-9fd2-24b1469ee2f3.png">

2) Can we do something to fix the cut off face image at the top? :)
<img width="1409" alt="Screen Shot 2019-07-04 at 12 32 04 PM" src="https://user-images.githubusercontent.com/2958442/60685433-f6b65a80-9e57-11e9-9529-6e59b978980e.png">

gkalpak committed:
fix(docs-infra): fix layout of `file-not-found` page

The `.sidenav-content` element, [which is also][1] the `#file-not-found`
element, [already has][2] the necessary padding. Especially the top
padding is important to ensure the element is not obscured by the
top-bar.

[1]: https://github.com/angular/angular/blob/1d3e22766/aio/src/app/app.component.html#L51
[2]: https://github.com/angular/angular/blob/1d3e22766/aio/src/styles/1-layouts/_content-layout.scss#L10

mary-poppins commented:
You can preview a9b3ede at https://pr31390-a9b3ede.ngbuilds.io/.

gkalpak commented:
Thx for taking a look, @sjtrimble. I fixed the image placement.

Regarding the "no results" message, I wouldn't be opposed to showing something (although I assume it should rarely come up for the `page-not found` error page). It is not related to this PR though (I am just fixing the color of the message to be different from the background and thus visible). Feel free to open an issue (or a PR :wink:) about changing the message and we can discuss the details there.

sjtrimble commented:
One more thing @gkalpak .. I'm still seeing the headers showing up a white and thereby transparent. Try seeing here: https://pr31390-119dcd1.ngbuilds.io/ngmodules

sjtrimble commented on aio/src/styles/2-modules/_search-results.scss:
```diff
@@ -20,81 +18,84 @@ aio-search-results {
+          @include font-size(14);
+          color: $lightgray;
+          font-weight: normal;
+          &a {
```
I normally put a line break here so it's easier to read ... if you feel like it :)

gkalpak replied:
I was keeping inline with the convention in this file, which seems to be "no line break, if it's targeting the same element". Happy to go either way.

gkalpak replied:
I added the line breaks (since I agree it is more readable).

gkalpak committed:
fixup! refactor(docs-infra): clean up `aio-search-results` styles

gkalpak commented:
> I'm still seeing the headers showing up a white and thereby transparent.

Great catch! 😱 😱 😱
Fixed.

mary-poppins commented:
You can preview 4be1fe9 at https://pr31390-4be1fe9.ngbuilds.io/.

mary-poppins commented:
You can preview 9967ab9 at https://pr31390-9967ab9.ngbuilds.io/.

mary-poppins commented:
You can preview 4fac6dd at https://pr31390-4fac6dd.ngbuilds.io/.
